src/dune/watermarks.ml

```diff
@@ -310,17 +346,27 @@ let subst vcs =
           (fun () -> Vcs.commit_id vcs))
       (fun () -> Vcs.files vcs)
   in
+  let files_ = String.Set.of_list (List.map ~f:Path.to_string files) in
```

Comment:
`files_` is not super explicit as a variable name, I propose to call it just `files` and move it under `let dune_project`, i.e.:

```ocaml
  let dune_project =
    let files = ... in
```